modules/apps/portal-workflow/portal-workflow-kaleo-test/src/testIntegration/java/com/liferay/portal/workflow/kaleo/change/tracking/test/KaleoTimerInstanceTokenTableReferenceDefinitionTest.java

```diff
@@ -70,12 +79,40 @@ public void setUp() throws Exception {
 
 	@Override
 	protected CTModel<?> addCTModel() throws Exception {
-		return _kaleoTimerInstanceTokenLocalService.addKaleoTimerInstanceToken(
-			_kaleoInstanceToken.getKaleoInstanceTokenId(),
-			_kaleoTaskInstanceToken.getKaleoTaskInstanceTokenId(),
-			_kaleoTimer.getKaleoTimerId(), _kaleoTimer.getName(),
-			WorkflowContextUtil.convert(_kaleoInstance.getWorkflowContext()),
-			serviceContext);
+		KaleoTimerInstanceToken kaleoTimerInstanceToken =
+			_kaleoTimerInstanceTokenLocalService.addKaleoTimerInstanceToken(
+				_kaleoInstanceToken.getKaleoInstanceTokenId(),
+				_kaleoTaskInstanceToken.getKaleoTaskInstanceTokenId(),
+				_kaleoTimer.getKaleoTimerId(), _kaleoTimer.getName(),
+				WorkflowContextUtil.convert(
+					_kaleoInstance.getWorkflowContext()),
+				serviceContext);
+
+		_executeTimer(kaleoTimerInstanceToken);
+
+		return _kaleoTimerInstanceTokenLocalService.
+			fetchKaleoTimerInstanceToken(
+				kaleoTimerInstanceToken.getKaleoTimerInstanceTokenId());
+	}
+
+	private void _executeTimer(KaleoTimerInstanceToken kaleoTimerInstanceToken)
+		throws Exception {
+
+		String schedulerGroupName = SchedulerUtil.getGroupName(
+			kaleoTimerInstanceToken.getCompanyId(),
+			kaleoTimerInstanceToken.getKaleoTimerInstanceTokenId());
+
+		SchedulerResponse schedulerResponse =
+			SchedulerEngineHelperUtil.getScheduledJob(
+				schedulerGroupName, schedulerGroupName, StorageType.PERSISTED);
+
+		Message message = schedulerResponse.getMessage();
+
+		message.put(
+			SchedulerEngine.DESTINATION_NAME,
+			KaleoRuntimeDestinationNames.WORKFLOW_TIMER);
```

Comment:
Is this needed? The DESTINATION_NAME is used by the scheduler to determine which message listener to call. Since we're passing the message directly to the message listener, this seems unnecessary to me.